src/typescript-generator/operations-type-coder.js

```diff
@@ -0,0 +1,130 @@
+import nodePath from "node:path";
+
+import { CONTEXT_FILE_TOKEN } from "./context-file-token.js";
+import { ParametersTypeCoder } from "./parameters-type-coder.js";
+import { READ_ONLY_COMMENTS } from "./read-only-comments.js";
+import { ResponsesTypeCoder } from "./responses-type-coder.js";
+import { SchemaTypeCoder } from "./schema-type-coder.js";
+import { TypeCoder } from "./type-coder.js";
+
+export class OperationsTypeCoder extends TypeCoder {
+  constructor(requirement, requestMethod) {
+    super(requirement);
+
+    if (requestMethod === undefined) {
+      throw new Error("requestMethod is required");
+    }
+
+    this.requestMethod = requestMethod;
+  }
+
+  names() {
+    return super.names(`HTTP_${this.requestMethod.toUpperCase()}`);
+  }
+
+  responseTypes(script) {
+    return this.requirement
+      .get("responses")
+      .flatMap((response, responseCode) => {
+        const status =
+          responseCode === "default"
+            ? "number | undefined"
+            : Number.parseInt(responseCode, 10);
+
+        if (response.has("content")) {
+          return response.get("content").map(
+            (content, contentType) => `{  
+              status: ${status}, 
+              contentType?: "${contentType}",
+              body?: ${new SchemaTypeCoder(content.get("schema")).write(script)}
+            }`,
+          );
+        }
+
+        if (response.has("schema")) {
+          const produces =
+            this.requirement?.get("produces")?.data ??
+            this.requirement.specification.rootRequirement.get("produces").data;
+
+          return produces
+            .map(
+              (contentType) => `{
+            status: ${status},
+            contentType?: "${contentType}",
+            body?: ${new SchemaTypeCoder(response.get("schema")).write(script)}
+          }`,
+            )
+            .join(" | ");
+        }
+
+        return `{  
+          status: ${status} 
+        }`;
+      })
+
+      .join(" | ");
+  }
+
+  modulePath() {
+    const pathString = this.requirement.url
+      .split("/")
+      .at(-2)
+      .replaceAll("~1", "/");
+
+    return `${nodePath
+      .join("path-types", pathString)
+      .replaceAll("\\", "/")}.types.ts`;
+  }
+
+  // eslint-disable-next-line max-statements
+  writeCode(script) {
+    // eslint-disable-next-line no-param-reassign
+    script.comments = READ_ONLY_COMMENTS;
+
+    const xType = script.importSharedType("WideOperationArgument");
```

Comment:
Suggest using an `enum` instead of hard coded strings:
```suggestion
    const xType = script.importSharedType(ShareType.WideOperationArgument);
```

Reply:
These strings appear as both a type name and a runtime value. I don't think there's a way to share across those two different contexts. Otherwise good idea.